oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -41,51 +48,65 @@ class ElasticIndexHelper {
-     *
-     * TODO: index create cannot be migrated to the ES Java client: it does not support custom mappings/settings needed to configure elastiknn.
-     * See discussion in https://discuss.elastic.co/t/elasticsearch-java-client-support-for-custom-mappings-settings/303172
-     * The migration will continue when this roadmap item gets fixed https://github.com/elastic/elasticsearch-java/issues/252
      */
-    public static CreateIndexRequest createIndexRequest(String remoteIndexName, ElasticIndexDefinition indexDefinition) throws IOException {
-        final CreateIndexRequest request = new CreateIndexRequest(remoteIndexName);
-
-        // provision settings
-        request.settings(loadSettings(indexDefinition));
+    public static CreateIndexRequest createIndexRequest(@NotNull String remoteIndexName,
+                                                        @NotNull ElasticIndexDefinition indexDefinition) {
+        return new CreateIndexRequest.Builder()
+                .index(remoteIndexName)
+                .settings(s -> loadSettings(s, indexDefinition))
+                .mappings(s -> loadMappings(s, indexDefinition))
+                .build();
+    }
 
-        // provision mappings
-        final XContentBuilder mappingBuilder = XContentFactory.jsonBuilder();
-        mappingBuilder.startObject();
-        {
-            mappingBuilder.startObject("properties");
-            {
-                mapInternalProperties(mappingBuilder);
-                mapIndexRules(indexDefinition, mappingBuilder);
-            }
-            mappingBuilder.endObject();
-        }
-        mappingBuilder.endObject();
-        request.mapping(mappingBuilder);
+    private static ObjectBuilder<TypeMapping> loadMappings(@NotNull TypeMapping.Builder builder,
+                                                           @NotNull ElasticIndexDefinition indexDefinition) {
+        mapInternalProperties(builder);
+        mapIndexRules(builder, indexDefinition);
+        return builder;
+    }
 
-        return request;
+    private static void mapInternalProperties(@NotNull TypeMapping.Builder builder) {
+        builder.properties(FieldNames.PATH,
+                        b1 -> b1.keyword(builder3 -> builder3))
+                .properties(FieldNames.ANCESTORS,
+                        b1 -> b1.text(
+                                b2 -> b2.analyzer("ancestor_analyzer")
+                                        .searchAnalyzer("keyword")
+                                        .searchQuoteAnalyzer("keyword")))
+                .properties(FieldNames.PATH_DEPTH,
+                        b1 -> b1.integer(
+                                b2 -> b2.docValues(false)))
+                .properties(FieldNames.FULLTEXT,
+                        b1 -> b1.text(
+                                b2 -> b2.analyzer("oak_analyzer")));
+        // TODO: the mapping below is for features currently not supported. These need to be reviewed
```

Comment:
would it make sense to create an OAK ticket for it?

Reply:
This TODO was already in the code, I just moved it around. I don't have the context to understand what it means by these features currently not supported. The comment below was originally written by @nit0906 (or at least, it's what git blame indicates). @nit0906 , do you remember what this is about?